tests/neg-custom-args/fatal-warnings/enum-variance.check

```diff
@@ -0,0 +1,5 @@
+-- Error: tests/neg-custom-args/fatal-warnings/enum-variance.scala:2:12 ------------------------------------------------
+2 |  case Refl(f: T => T) // error: covariant T in contravariant position... enum case may require explicit type parameters
+  |            ^^^^^^^^^
+  |            covariant type T occurs in contravariant position in type T => T of value f
+  |            enum case class Refl may require explicit type parameters to resolve this issue
```

Comment:
- Can we tweak the error message so this line is only displayed when a) the case has no type parameters and b) the type used in the wrong position is a type parameter of the enclosing enum? That way we should be able to drop the "may" from the message.
- I think one of the documentation page on enum should explain this issue and how to fix it with an example, then we can link to that example from this error message.

Reply:
I've added the link, but I wasn't sure if there was any better check than position of spans

Reply:
changed to synthetic flag check on tear